src/Microsoft.AspNet.Server.Kestrel/Http/Frame.cs

```diff
@@ -744,7 +751,50 @@ private bool TakeStartLine(SocketInput input)
                 RequestUri = requestUrlPath;
                 QueryString = queryString;
                 HttpVersion = httpVersion;
-                Path = RequestUri;
+
+                var pathBaseEnd = pathBegin;
```

Comment:
This is not the final revision of it - I just pushed so I can continue working on it later from somewhere else.